dcrec/secp256k1/privkey.go

```diff
@@ -62,6 +67,25 @@ func GeneratePrivateKey() (*PrivateKey, error) {
 	return &key, nil
 }
 
+// GeneratePrivateKey generates and returns a new cryptographically secure
+// private key that is suitable for use with secp256k1.
+func GeneratePrivateKey() (*PrivateKey, error) {
+	return generatePrivateKey(nil)
+}
+
+// GenerateKey generates a public/private key pair using entropy from rand.  If
+// rand is nil, [crypto/rand.Reader] will be used.
+func GenerateKey(rand io.Reader) (*PublicKey, *PrivateKey, error) {
```

Comment:
We could reverse the order.  Public and private in that order just seems more natural.  Nobody refers to a "private and public" key pair, it's is always "public and private".

The function name just says Key, without specifying public or private.  I wouldn't assume any particular order based on just the name alone.

The arguments are also different types; it would be difficult to use them incorrectly.